Userland/Libraries/LibJS/Bytecode/Pass/MergeBlocks.cpp

```diff
@@ -48,6 +48,14 @@ void MergeBlocks::perform(PassPipelineExecutable& executable)
                 continue;
         }
 
+        {
+            InstructionStreamIterator it { entry.key->instruction_stream() };
+            while (!(*it).is_terminator())
+                ++it;
+            if ((*it).type() != Instruction::Type::Jump)
+                continue;
+        }
+
```

Comment:
This seems insanely expensive, can we instead let the block hold its terminator type?

Reply:
Done in a hopefully nice manner, and moved that check up a bit